ablySpec/RealtimeClientChannel.swift

```diff
@@ -1114,6 +1114,40 @@ class RealtimeClientChannel: QuickSpec {
 
             }
 
+            context("channels") {
+                // RTS4
+                context("release") {
+                    it("should release a channel") {
+                        let client = ARTRealtime(options: AblyTests.commonAppSetup())
+                        defer { client.close() }
+
+                        let channel = client.channels.get("test")
+                        channel.subscribe { _ in
+                            fail("shouldn't happen")
+                        }
+                        channel.presence.subscribe { _ in
+                            fail("shouldn't happen")
+                        }
+                        waitUntil(timeout: testTimeout) { done in
+                            client.channels.release("test") { errorInfo in
+                                expect(errorInfo).to(beNil())
+                                expect(channel.state).to(equal(ARTRealtimeChannelState.Detached))
+                                done()
+                            }
+                        }
+
+                        let sameChannel = client.channels.get("test")
+                        waitUntil(timeout: testTimeout) { done in
+                            sameChannel.subscribe { _ in
+                                sameChannel.presence.enterClient("foo", data: nil) { _ in
+                                    delay(0.0) { done() } // Delay to make sure EventEmitter finish its cycle.
```

Comment:
Yes, the delay is just to move the block to an "immediate event" in the event loop, so that we make sure that the presence subscription above gets called before `done()` is called in this line in case `release` doesn't work, ie. doesn't unsubscribe it.

Maybe there's a less confusing way of doing this...